File: Suma de enteros/Suma de enteros.pas
program Hd;

var
    num,n,aux,p:Int64;

begin
    Readln(num);
    n:=Trunc(sqrt(num div 2));
    while n<num do
    begin
        aux:= (n*(n+1)) div 2;
        if aux = num then
        begin
            Writeln(0);
            break
        end    
        else if num-aux<=n then
        begin
            Writeln(n,' ',num-aux);
            break
        end;
        n:=n+1;
    end;
end.

File: Suma de enteros/Suma de enteros.cpp
#include<bits/stdc++.h>
using namespace std;
#define Hd main(void)


Hd
{
    ios::sync_with_stdio(0);
    cin.tie(nullptr);
    cout.tie(nullptr);
    long long num;
    cin>>num;
    long long n = sqrt(num/2);
    for(;n<num;n++){
        long long aux  = (n*(n+1))/2;
        if(aux == num){
            cout<<0;
            return 0;
            }
        else{
            if(aux<num){
                if(num-aux <= n){cout<<n<<" "<<num-aux<<"\n";return 0;}
            }
        }
    }
}